tools/tck-reporting/src/main/java/org/opencypher/tools/tck/reporting/SystemOutReader.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2015-2019 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Attribution Notice under the terms of the Apache License 2.0
+ *
+ * This work was created by the collective efforts of the openCypher community.
+ * Without limiting the terms of Section 6, any Derivative Work that is not
+ * approved by the public consensus process of the openCypher Implementers Group
+ * should not be described as “Cypher” (and Cypher® is a registered trademark of
+ * Neo4j Inc.) or as "openCypher". Extensions by implementers or prototypes or
+ * proposals for change that have been documented or implemented should only be
+ * described as "implementation extensions to Cypher" or as "proposed changes to
+ * Cypher that are not yet approved by the openCypher community".
+ */
+package org.opencypher.tools.tck.reporting;
+
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+
+class SystemOutReader {
+    private PrintStream systemOut = System.out;
+    private ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+    private TeeOutputStream teeOutput = new TeeOutputStream(System.out, buffer);
+
+    SystemOutReader() {
+        System.setOut(new PrintStream(teeOutput));
```

Comment:
Is there a nicer way to achieve the same other than changing the default output stream?

I wonder how this behaves when different test classes are run in parallel.

Reply:
1. My understanding is that TCK is not designed to be executed in parallel.
2. Capturing the output stream is the most universal approach. Otherwise, it would require implementer to forward logs to the adapter, which is not very straight forward approach.
3. I was unaware of [new experimental output capturing functionality](https://junit.org/junit5/docs/current/user-guide/#running-tests-capturing-output) in JUnit 5.3, will look into it.

Reply:
JUnit output capturing functionality, will not work in this case because the adapter is based on TCK events, not JUnit events.

I can't think of an approach that will be nicer than changing the default output stream. Maybe you have any ideas?